zha/application/platforms/climate/__init__.py

```diff
@@ -589,6 +589,67 @@ class ZenWithinThermostat(Thermostat):
     """Zen Within Thermostat implementation."""
 
 
+@MULTI_MATCH(
+    cluster_handler_names=CLUSTER_HANDLER_THERMOSTAT,
+    manufacturers={"ZEHNDER GROUP VAUX ANDIGNY      "},
```

Comment:
Hi,
I just tested this branch on my Home Assistant.

Personally, my heater's manufacturer name does not include the trailing spaces. Maybe I have a more recent version, and ZEHNDER has fixed this typo?

I believe both manufacturer names, "ZEHNDER GROUP VAUX ANDIGNY " and "ZEHNDER GROUP VAUX ANDIGNY", exist on the market.

Therefore, I propose a PR to @gdoffe repository to allow support for both manufacturer names. I hope it will be accepted and reflected in this PR :)

Otherwise, I tested it on my side (with this minor change), and it works very well. The modes are correctly mapped thanks to @gdoffe work.